app/services/dashboard_service.rb

```diff
@@ -39,4 +39,14 @@ def format_memory(used, total)
   def mem_num(val)
     ((val || 0) / 1024.0)
   end
+
+  def get_url(ems_id, attr_url)
+    ems_id + "?display=" + attr_url
+  end
+
+  def get_icon(ems_type)
+    type = ems_type.split('::')
+    vendor = type[2].downcase
+    ActionController::Base.helpers.image_path('svg/vendor-' + vendor + '.svg')
```

Comment:
Ok, this will work, but if possible, we should be using decorators to get the right filename..

Am I right in assuming `ems_type` is the entire classname as a string?

If so, `ActionController::Base.helpers.image_path(ems_type.constantize.decorate.fileicon)` should work.
(But you may need to make sure the decorator for each class you support (found under `app/decorators/(class_name_or_a_parent_class).rb`) does have a `self.fileicon` method implemented.)